addons/arsenal/functions/fnc_updateRightPanel.sqf

```diff
@@ -1,57 +1,56 @@
 #include "script_component.hpp"
 #include "..\defines.hpp"
 /*
- * Author: Alganthe
+ * Author: Alganthe, johnb43
  * Update the right panel (listnbox).
  *
  * Arguments:
  * 0: Right panel control <CONTROL>
- * 1: Max load of the current container <STRING>
+ * 1: Container <OBJECT>
+ * 2: If container has items <BOOL>
  *
  * Return Value:
  * None
  *
  * Public: No
 */
 
-params ["_control", "_maxLoad"];
+params ["_control", "_container", "_hasItems"];
 
-private _loadIndicatorBarCtrl = _display displayCtrl IDC_loadIndicatorBar;
-private _curSel = lnbCurSelRow _control;
-
-(lnbSize _control) params ["_rows"];
+private _loadRemaining = maxLoad _container - loadAbs _container;
 
-_maxLoad = getnumber (configfile >> "CfgVehicles" >> _maxLoad >> "maximumLoad");
-_maxLoad = _maxLoad * (1 - (progressPosition _loadIndicatorBarCtrl));
-_maxLoad = parseNumber (_maxLoad toFixed 2); // Required to avoid an issue where even though the typename returns "SCALAR" it doesn't act as one.
```

Comment:
`_display` was never defined anywhere, so that's why he kept getting `SCALAR`.

Reply:
But if it was undefined/nil, it wouldn't be able to be passed to typeName 🤔